forc-plugins/forc-id/src/op/contract_id.rs

```diff
@@ -0,0 +1,75 @@
+use crate::cmd::contract_id::Command as ContractIdCommand;
+use anyhow::{bail, Result};
+use forc_pkg::{self as pkg, build_with_options};
+use sway_core::{fuel_prelude::fuel_tx, BuildTarget};
+use tracing::info;
+
+pub fn contract_id(command: ContractIdCommand) -> Result<()> {
+    let build_options = build_opts_from_cmd(&command);
+    let build_plan = pkg::BuildPlan::from_build_opts(&build_options)?;
+    // If a salt was specified but we have more than one member to build, there
+    // may be ambiguity in how the salt should be applied, especially if the
+    // workspace contains multiple contracts, and especially if one contract
+    // member is the dependency of another (in which case salt should be
+    // specified under `[contract- dependencies]`). Considering this, we have a
+    // simple check to ensure that we only accept salt when working on a single
+    // package. In the future, we can consider relaxing this to allow for
+    // specifying a salt for workspacs, as long as there is only one
+    // root contract member in the package graph.
+    if command.salt.salt.is_some() && build_plan.member_nodes().count() > 1 {
+        bail!(
+            "A salt was specified when attempting to contract id detection \
```

Comment:
```suggestion
            "A salt was specified when attempting to detect the contract id \
```

Also the behaviour for deploy in workspaces will change because of the issue i'm working on now (#4117). Perhaps this could be a follow-up issue to that.